src/moments.jl

```diff
@@ -221,19 +221,19 @@ end
 
 
 ##### General central moment
-function _moment2(v::RealArray, m::Real; corrected=false)
+function _moment2(v::RealArray{T}, m::Real; corrected=false) where T
```

Comment:
Rather than forcing specialization, I would just replace the use of `T` below with `eltype(v)`. Same for all of the other methods here.

Reply:
“Forcing specialization”? Isn’t Julia always specializing?

Reply:
@ararslan that's how I originally had it, but I wanted to get some additional type stability because I figured that moments might be something in a tight loop. It turned out harder to have that specialization on the weight vector (which is why it's not specialized) because the type parameterization is complicated and nested.

@mschauer I think the trick here is that it's parametric types and how much specialization is done on the type parameter.  

I'm happy to have the simpler variant with `eltype`.